src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -498,6 +498,7 @@ def process_messages(
                 headers=[
                     *message.payload.headers,
                     ("mapping_sources", mapping_header_content),
+                    ("metric_type", parsed_payload_value["type"]),
```

Comment:
I think we should validate the content of the type field in the indexer instead of just trusting the input. 
That makes snuba depend on the contract it has with the indexer without having to deal with the original format produced by relay. Thus reduces the set of assumptions snuba has to care for. 
(And drop invalid messages)

Reply:
do you want a new enum type here so that it gets constrained everywhere or are you OK with just a check in the original parsing/tag extraction step e.g. https://github.com/getsentry/sentry/blob/30d69d8c48fcd2eabc822a79c312f0a5d6a50f82/src/sentry/sentry_metrics/multiprocess.py#L406-L412